app.py

```diff
@@ -42,19 +41,31 @@ def ocrmypdf_api():
     if uploaded_file.filename == "":
         return "No file selected", 400
 
-    if uploaded_file.mimetype != "application/pdf":
-        return "Invalid file type, please upload a PDF file", 400
+    allowed_extensions = {".pdf", ".jpg", ".jpeg", ".png"}
+    file_extension = os.path.splitext(uploaded_file.filename)[1]
+
+    if file_extension not in allowed_extensions:
+        return "Invalid file type, please upload a PDF or image file", 400
+
+    if file_extension in {".jpg", ".jpeg", ".png"}:
+    # convert the image to PDF
+        with Image.open(uploaded_file) as img:
+            with tempfile.NamedTemporaryFile(suffix=".pdf", delete=False) as temp_file:
+                img.convert("RGB").save(temp_file, "PDF")
+            pdf_file = open(temp_file.name, 'rb').read()
+    else:
+        pdf_file = uploaded_file.read() 
```

Comment:
`ocrmypdf` supports image natively, I would try through that route before resorting to pillow/PIL